moveit_planners/ompl/ompl_interface/test/test_planning_context_manager.cpp

```diff
@@ -0,0 +1,304 @@
+ *
+ * TODO(jeroendm) I also tried something similar with position constraints, but get a segmentation fault
+ * that occurs in the 'geometric_shapes' package, in the method 'useDimensions' in 'bodies.h'.
+ * git permalink:
+ * https://github.com/ros-planning/geometric_shapes/blob/df0478870b8592ce789ee1919f3124058c4327d7/include/geometric_shapes/bodies.h#L196
+ *
+ **/
+
+#include "load_test_robot.h"
+
+#include <gtest/gtest.h>
+
+#include <eigen_conversions/eigen_msg.h>
+
+#include <moveit/ompl_interface/planning_context_manager.h>
+#include <moveit/planning_scene/planning_scene.h>
+#include <moveit/planning_interface/planning_request.h>
+#include <moveit/robot_state/conversions.h>
+#include <moveit/kinematic_constraints/utils.h>
+#include <moveit/constraint_samplers/constraint_sampler_manager.h>
+
+#include <moveit/ompl_interface/parameterization/joint_space/joint_model_state_space.h>
+
+/** \brief Generic implementation of the tests that can be executed on different robots. **/
+class TestPlanningContext : public ompl_interface_testing::LoadTestRobot, public testing::Test
+{
+public:
+  TestPlanningContext(const std::string& robot_name, const std::string& group_name)
+    : LoadTestRobot(robot_name, group_name)
+  {
+  }
+
+  // /***************************************************************************
+  //  * START Test implementations
+  //  * ************************************************************************/
+
+  void testSimpleRequest(const std::vector<double>& start, const std::vector<double>& goal)
+  {
+    // create all the test specific input necessary to make the getPlanningContext call possible
+    planning_interface::PlannerConfigurationSettings pconfig_settings;
+    pconfig_settings.group = group_name_;
+    pconfig_settings.name = group_name_;
+    pconfig_settings.config = { { "enforce_joint_model_state_space", "0" }, { "use_ompl_constrained_planning", "0" } };
+
+    planning_interface::PlannerConfigurationMap pconfig_map{ { pconfig_settings.name, pconfig_settings } };
+    moveit_msgs::MoveItErrorCodes error_code;
+    planning_interface::MotionPlanRequest request = createRequest(start, goal);
+
+    // setup the planning context manager
+    ompl_interface::PlanningContextManager pcm(robot_model_, constraint_sampler_manager_);
+    pcm.setPlannerConfigurations(pconfig_map);
+
+    // see if it returns the expected planning context
+    auto pc = pcm.getPlanningContext(planning_scene_, request, error_code, node_handle_, false);
+
+    EXPECT_NE(pc->getOMPLSimpleSetup(), nullptr);
+    auto ss = dynamic_cast<ompl_interface::JointModelStateSpace*>(pc->getOMPLStateSpace().get());
+    EXPECT_NE(ss, nullptr);
+
+    // solve the planning problem
+    planning_interface::MotionPlanDetailedResponse res;
+    bool success = pc->solve(res);
+    EXPECT_TRUE(success);
+  }
+
+  void testPathConstraints(const std::vector<double>& start, const std::vector<double>& goal)
```

Comment:
I added tests to see if the path constraints are added to the planning context and properly used by the planner. (The specific start and goal state used maybe a bit too trivial, but I'll leave this as future work if that's ok. Otherwise, I fear I might never finish this PR and continue working on the constrained planning PR.)
```C++
    // Are the path constraints created in the planning context?
    auto path_constraints = pc->getPathConstraints();
    EXPECT_FALSE(path_constraints->empty());
    EXPECT_EQ(path_constraints->getOrientationConstraints().size(), 1);
    EXPECT_TRUE(path_constraints->getPositionConstraints().empty());
    EXPECT_TRUE(path_constraints->getJointConstraints().empty());
    EXPECT_TRUE(path_constraints->getVisibilityConstraints().empty());

    // Check if all the states in the solution satisfy the path constraints.
    // A detailed response returns 3 solutions: the ompl solution, the simplified solution and the interpolated
    // solution. We test all of them here.
    kinematic_constraints::ConstraintEvaluationResult con_result;
    for (const robot_trajectory::RobotTrajectoryPtr& trajectory : res.trajectory_)
    {
      for (std::size_t pt_index = { 0 }; pt_index < trajectory->getWayPointCount(); ++pt_index)
      {
        EXPECT_TRUE(path_constraints->decide(trajectory->getWayPoint(pt_index)).satisfied);
      }
    }
```